Quote all first characters the Dataverse cannot handle

christian-monch opened the pull request:
Datavers has the peculiar habit of removing leading ``.``, ``-``, and `` `` from file names. This PR ensures that all three characters are escaped, if they are at the start of a file name. Previously only ``.`` was escaped.

christian-monch committed:
quote all forbidden first chars for dataverse

Datavers has the peculiar habit of removing
leading '.', '-', and ' ' from file names.
This commit ensures that all three characters
are escaped, if they are at the start of a
file name. Previously only "." was escaped.

mih commented:
Thx! I will wait till #273 is tested and merge, then rerun the tests here, and then merge.

mih commented:
In general, it is nice that this is all self-contained and tested within. However, it would be good to have an explicit test that actually puts a problematic file on dataverse and gets it back.

Ping #202

christian-monch committed:
add name mangling tests against dataverse instance

This PR adds tests that use "difficult" names in
combination with a real dataverse instance

mih reviewed (approved):
Thx for the fix!
